app/controllers/vms_controller.rb

```diff
@@ -56,9 +54,7 @@ def update
     notify_changed_users(@vm.users.map(&:id), info_params[:non_sudo_user_ids].map(&:to_i), false, @vm.name)
     @vm.sudo_users = info_params[:sudo_user_ids]
     @vm.users = info_params[:non_sudo_user_ids]
-    @vm.ensure_config.description = info_params[:description]
-    @ip = @vm.ensure_config.ip = info_params[:ip]
-    @dns = @vm.ensure_config.dns = info_params[:dns]
+    @vm.ensure_config.update config_params
     unless @vm.config.save
```

Comment:
The update method already does a save to the database and returns whether it was successful.
So use the update in the unless statement directly, instead of first updating, then saving.